1.12.0 Alpine build fail Issue with mbedtls

Issue occurs in ---

src/server.c: In function 'Server_init_rng':
src/server.c:176:29: error: 'mbedtls_havege_state' undeclared (first use in this function); did you mean 'mbedtls_sha512_starts'?
         ctx = calloc(sizeof(mbedtls_havege_state), 1);
                             ^~~~~~~~~~~~~~~~~~~~
                             mbedtls_sha512_starts
src/server.c:176:29: note: each undeclared identifier is reported only once for each function it appears in
src/server.c:177:9: warning: implicit declaration of function 'mbedtls_havege_init'; did you mean 'mbedtls_aes_init'? [-Wimplicit-function-declaration]
         mbedtls_havege_init((mbedtls_havege_state *)ctx);
         ^~~~~~~~~~~~~~~~~~~
         mbedtls_aes_init
src/server.c:177:52: error: expected expression before ')' token
         mbedtls_havege_init((mbedtls_havege_state *)ctx);
                                                    ^
src/server.c:179:25: error: 'mbedtls_havege_random' undeclared (first use in this function); did you mean 'mbedtls_ctr_drbg_random'?
         srv->rng_func = mbedtls_havege_random;
                         ^~~~~~~~~~~~~~~~~~~~~
                         mbedtls_ctr_drbg_random
make: *** [<builtin>: src/server.o] Error 1

---

Trying to build this using alpine vs 3.9 with 
make
gcc
g++
zeromq
zeromq-dev
sqlite
sqlite-dev
mbedtls-dev
mbedtls-utils
openssh
git
tar
curl

# curl -L https://github.com/mongrel2/mongrel2/releases/download/v1.12.0/mongrel2-v1.12.0.tar.bz2 | tar --strip-components=1 -xj -C m2 && cd m2 && make all install

If you would like I can add a copy of the Dockerfile i am writing to build this.

Thanks

--- a/src/server.c
+++ b/src/server.c
@@ -171,6 +171,7 @@ int Server_init_rng(Server *srv)
         srv->rng_func = mbedtls_ctr_drbg_random;
         srv->rng_ctx = ctx;
     } else {
+#ifdef MBEDTLS_HAVEGE_C
         log_warn("entropy source unavailable. falling back to havege rng");
 
         ctx = calloc(sizeof(mbedtls_havege_state), 1);
@@ -178,6 +179,9 @@ int Server_init_rng(Server *srv)
 
         srv->rng_func = mbedtls_havege_random;
         srv->rng_ctx = ctx;
+#else
+	sentinel("No entropy source available, SSL is not possible (disable chroot?)\n");
+#endif
     }
 
     return 0;

